### PR TITLE
Upgrade to Aspetcore3.1

### PR DESCRIPTION
**Pull request by yashagarwal23**
- Updated Target Frameworks to net461; netcoreapp3.1; net5.0; net6.0
- Updated Tests to netcoreapp3.1
- Updated build.ps1 to use VS 2022 MSBuild

### CODE REVIEW

File: properties/service_fabric_managed.targets [hunk 1]
```diff
@@ -49,12 +49,37 @@
   <Target Name="CopyProdBinaries">
     <Message Text="Copying binaries to drop folder ..." Importance="High"/>
     <ItemGroup>
-      <BinairesNetStandard Include="src\Microsoft.ServiceFabric.AspNetCore\$(OutputPath)Microsoft.ServiceFabric.AspNetCore.*"/>
-      <BinairesNetStandard Include="src\Microsoft.ServiceFabric.AspNetCore.Configuration\$(OutputPath)Microsoft.ServiceFabric.AspNetCore.Configuration.*"/>
-      <BinairesNetStandard Include="src\Microsoft.ServiceFabric.AspNetCore.HttpSys\$(OutputPath)Microsoft.ServiceFabric.AspNetCore.HttpSys.*"/>
-      <BinairesNetStandard Include="src\Microsoft.ServiceFabric.AspNetCore.Kestrel\$(OutputPath)Microsoft.ServiceFabric.AspNetCore.Kestrel.*"/>
+      <BinairesNetFramework Include="src\Microsoft.ServiceFabric.AspNetCore\$(OutputPath)net461\Microsoft.ServiceFabric.AspNetCore.*"/>
```
Comment (PavanKunapareddyMSFT) [inline properties/service_fabric_managed.targets]: NIT: typo

File: src/Microsoft.ServiceFabric.AspNetCore.HttpSys/SR.Designer.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ namespace Microsoft.ServiceFabric.Services.Communication.AspNetCore {
     // class via a tool like ResGen or Visual Studio.
     // To add or remove a member, edit your .ResX file then rerun ResGen
     // with the /str option, or rebuild your VS project.
-    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "4.0.0.0")]
+    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "16.0.0.0")]
```
Comment (PavanKunapareddyMSFT) [inline src/Microsoft.ServiceFabric.AspNetCore.HttpSys/SR.Designer.cs:22]: I guess this change is from the generated code not the handmade one.

File: nuprojs/Microsoft.ServiceFabric.AspNetCore.Abstractions/Microsoft.ServiceFabric.AspNetCore.Abstractions.nuproj [hunk 3]
```diff
@@ -14,20 +14,133 @@
...
+      <Dependency Include="Microsoft.ServiceFabric.Services">
+        <Version>[$(NugetPkg_Version_Microsoft_ServiceFabric_Services)]</Version>
+        <TargetFramework>net5.0</TargetFramework>
+      </Dependency>
+      <FrameworkReference Include="Microsoft.AspNetCore.App">
+        <TargetFramework>net5.0</TargetFramework>
+      </FrameworkReference>
+    </ItemGroup>
+
+    <ItemGroup>
```
Comment (PavanKunapareddyMSFT) [inline nuprojs/Microsoft.ServiceFabric.AspNetCore.Abstractions/Microsoft.ServiceFabric.AspNetCore.Abstractions.nuproj]: You could probably use `<group>` to reduce verbosity

File: nuprojs/Microsoft.ServiceFabric.AspNetCore.Abstractions/Microsoft.ServiceFabric.AspNetCore.Abstractions.nuproj [hunk 4]
```diff
@@ -14,20 +14,133 @@
...
 
     <ItemGroup>
-      <File Include="$(DropFolderNetStandard)Microsoft.ServiceFabric.AspNetCore.dll">
-        <TargetPath>lib\netstandard2.0</TargetPath>
+      <File Include="$(DropFolderNetFramework)\Microsoft.ServiceFabric.AspNetCore.dll">
+        <TargetPath>lib\net461</TargetPath>
       </File>
-      <File Include="$(DropFolderNetStandard)Microsoft.ServiceFabric.AspNetCore.xml">
-        <TargetPath>lib\netstandard2.0</TargetPath>
+      <File Include="$(DropFolderNetFramework)\Microsoft.ServiceFabric.AspNetCore.xml">
```
Comment (PavanKunapareddyMSFT) [inline nuprojs/Microsoft.ServiceFabric.AspNetCore.Abstractions/Microsoft.ServiceFabric.AspNetCore.Abstractions.nuproj:21]: Net461 should be sufficient I believe. This nuproj is modelled this way to remain consistent with other nuprojs in winfab/actors repo
For versioning and backward compatibility - ref https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/version-compatibility#version-compatibility-for-components

**Review by PavanKunapareddyMSFT — APPROVED** [review]

File: properties/service_fabric_common.props [hunk 5]
```diff
@@ -12,7 +12,10 @@
     
     <!-- Set Drop folders -->
     <DropFolder>$(RepoRoot)drop\$(Configuration)\</DropFolder>
-    <DropFolderNetStandard>$(RepoRoot)drop\$(Configuration)\netstandard\</DropFolderNetStandard>
+    <DropFolderNetFramework>$(RepoRoot)drop\$(Configuration)\net461\</DropFolderNetFramework>
```
Comment (raunakpandya) [inline properties/service_fabric_common.props:15]: We dont need other framworks here?

File: build.ps1 [hunk 6]
```diff
@@ -33,36 +33,31 @@ if($MSBuildFullPath -ne "")
     }
 }
 
-# msbuild path not provided, find msbuild for VS2019
+# msbuild path not provided, find msbuild for VS2022
```
Comment (raunakpandya) [inline build.ps1:36]: Let's update readme of the repo to include the same.

**Review by raunakpandya — APPROVED** [review]

File: properties/service_fabric_managed.targets (hunk 1 above)
Comment (yashagarwal23) [reply properties/service_fabric_managed.targets]: fixed

File: build.ps1 (hunk 6 above)
Comment (yashagarwal23) [reply build.ps1:36]: Updated the Readme

File: src/Microsoft.ServiceFabric.AspNetCore.HttpSys/SR.Designer.cs (hunk 2 above)
Comment (yashagarwal23) [reply src/Microsoft.ServiceFabric.AspNetCore.HttpSys/SR.Designer.cs:22]: Yes, these changes are from generated code, not handmade

File: properties/service_fabric_common.props (hunk 5 above)
Comment (yashagarwal23) [reply properties/service_fabric_common.props:15]: Not required to create separate drop folders for each framework. The dlls in net461 drop folder will be copied to other net framework libs and runtimes folders while creating nupkg.

File: nuprojs/Microsoft.ServiceFabric.AspNetCore.Abstractions/Microsoft.ServiceFabric.AspNetCore.Abstractions.nuproj (hunk 3 above)
Comment (yashagarwal23) [reply nuprojs/Microsoft.ServiceFabric.AspNetCore.Abstractions/Microsoft.ServiceFabric.AspNetCore.Abstractions.nuproj]: ```<group>``` tag doesn't work in nuproj files

**Pull request merged**